src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/FrozenDictionary.AlternateLookup.cs

```diff
@@ -76,12 +76,17 @@ private protected IAlternateEqualityComparer<TAlternateKey, TKey> GetAlternateEq
         /// <typeparam name="TAlternateKey">The alternate type of a key for performing lookups.</typeparam>
         public readonly struct AlternateLookup<TAlternateKey> where TAlternateKey : notnull, allows ref struct
         {
+
+            private readonly AlternateLookupDelegate<TAlternateKey> _alternateLookupDelegate;
```

Comment:
It's probably ok in this case, but this does introduce a new failure mode, in that this struct is now multiple fields, which means it can be torn and you could end up with a struct containing a dictionary with a delegate that doesn't go with it.

Reply:
I will defer to you as to whether this is an acceptable risk as there is no way to get around it with the proposed strategy.